src/EventDispatcher/AbstractGuzzleHttpEvent.php

```diff
@@ -0,0 +1,198 @@
+<?php
+namespace M6Web\Bundle\GuzzleHttpBundle\EventDispatcher;
+
+use GuzzleHttp\Psr7\Request;
+use GuzzleHttp\Psr7\Response;
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\ResponseInterface;
+use Symfony\Contracts\EventDispatcher\Event;
+
+/**
+ * Class AbstractGuzzleHttpEvent
+ */
+abstract class AbstractGuzzleHttpEvent extends Event
+{
+    const EVENT_NAME = 'm6web.guzzlehttp';
+    const EVENT_ERROR_NAME = 'm6web.guzzlehttp.error';
+
+    /**
+     * Command start time
+     *
+     * @var float
+     */
+    protected $executionStart;
+
+    /**
+     * Command execution time
+     *
+     * @var float
+     */
+    protected $executionTime;
+
+    /**
+     * @var Request
+     */
+    protected $request;
+
+    /**
+     * @var Response
+     */
+    protected $response;
+
+    /**
+     * @var mixed
+     */
+    protected $reason;
+
+    /**
+     * @var string
+     */
+    protected $clientId;
+
+    /**
+     * Set request
+     *
+     * @param Request $request
+     *
+     * @return static
+     */
+    public function setRequest(RequestInterface $request)
```

Comment:
Is it an inherited method? Else what about return type?

```suggestion
    public function setRequest(RequestInterface $request): self
```